apps/src/blockly/customBlocks/googleBlockly/proceduresBlocks.js

```diff
@@ -233,30 +232,24 @@ GoogleBlockly.Extensions.registerMixin(
  * @param {WorkspaceSvg} workspace The workspace containing procedures.
  * @returns an array of block objects representing the flyout blocks
  */
-export function flyoutCategory(workspace, functionEditorOpen = false) {
+export function flyoutCategory(workspace) {
   const blockList = [];
 
-  // Note: Blockly.Msg was undefined when this code was extracted into global scope
-  const functionDefinitionBlock = {
-    kind: 'block',
-    type: BLOCK_TYPES.procedureDefinition,
-    fields: {
-      NAME: Blockly.Msg.PROCEDURES_DEFNORETURN_PROCEDURE,
-    },
-    extraState: {
-      userCreated: true,
-    },
-  };
-
-  if (functionEditorOpen) {
-    // No-op - cannot create new functions while the modal editor is open
-  } else if (Blockly.useModalFunctionEditor) {
-    const newFunctionButton = getNewFunctionButtonWithCallback(
-      workspace,
-      functionDefinitionBlock
-    );
+  if (Blockly.useModalFunctionEditor) {
+    const newFunctionButton = getNewFunctionButtonWithCallback(workspace);
     blockList.push(newFunctionButton);
   } else {
+    // Note: Blockly.Msg was undefined when this code was extracted into global scope
+    const functionDefinitionBlock = {
```

Comment:
nit that doesn't need to be addressed: Since this is a block "constant"/"template" that I would have extracted into a `constants` file if I could have, it could also have continued to live above where it's used, but this also makes sense and no need to revert!